native-sql-engine/core/src/main/scala/org/apache/spark/sql/execution/adaptive/ColumnarCustomShuffleReaderExec.scala

```diff
@@ -48,11 +48,11 @@ case class ColumnarCustomShuffleReaderExec(
         partitionSpecs.map(_.asInstanceOf[PartialMapperPartitionSpec].mapIndex).toSet.size ==
           partitionSpecs.length) {
       child match {
-        case ShuffleQueryStageExec(_, s: ColumnarShuffleExchangeAdaptor) =>
+        case ShuffleQueryStageExec(_, s: ColumnarShuffleExchangeAdaptor, _) =>
```

Comment:
Refactor the code to fix compatibility issues. 
Also keep shim layer approach, but commented.